src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -730,19 +744,24 @@ private VendorSupplierDiversity createVendorSupplierDiversityForDropDownData (St
     }
     
     private List<VendorSupplierDiversity> buildVendorDiversitiesFromPmwFormCheckboxes(PaymentWorksVendor pmwVendor, Map<String, SupplierDiversity> paymentWorksToKfsDiversityMap, List<VendorSupplierDiversity> kfsVendorSupplierDiversities) {
-        ArrayList<String> pmwDiversities = new ArrayList<String>(Arrays.asList(pmwVendor.getDiversityClassifications().split("\\s*,\\s*")));
-        if (!pmwDiversities.isEmpty()) {
-             for (String pmwDiversity : pmwDiversities) {
-                 if (paymentWorksToKfsDiversityMap.containsKey(pmwDiversity)) {
-                     SupplierDiversity supplierDiversityFromMap = paymentWorksToKfsDiversityMap.get(pmwDiversity);
-                     kfsVendorSupplierDiversities.add(buildVendorSupplierDiversity(supplierDiversityFromMap.getVendorSupplierDiversityCode(), addOneYearToDate(getDateTimeService().getCurrentDate())));
-                 }
-                 else
-                 {
-                     LOG.info("buildVendorDiversities:: PaymentWorks Vendor : " + pmwVendor.getRequestingCompanyLegalName() + 
-                              "  Diversity Value : " + pmwDiversity + " does not have corresponding KFS SupplierDiversity defined."); 
+        if (paymentWorksFormModeService.shouldUseLegacyFormProcessingMode()) {
+            ArrayList<String> pmwDiversities = new ArrayList<String>(Arrays.asList(pmwVendor.getDiversityClassifications().split("\\s*,\\s*")));
+            if (!pmwDiversities.isEmpty()) {
+                 for (String pmwDiversity : pmwDiversities) {
+                     if (paymentWorksToKfsDiversityMap.containsKey(pmwDiversity)) {
+                         SupplierDiversity supplierDiversityFromMap = paymentWorksToKfsDiversityMap.get(pmwDiversity);
+                         kfsVendorSupplierDiversities.add(buildVendorSupplierDiversity(supplierDiversityFromMap.
+                                 getVendorSupplierDiversityCode(), addOneYearToDate(getDateTimeService().getCurrentDate())));
+                     }
+                     else
+                     {
+                         LOG.info("buildVendorDiversities:: PaymentWorks Vendor : " + pmwVendor.getRequestingCompanyLegalName() + 
+                                  "  Diversity Value : " + pmwDiversity + " does not have corresponding KFS SupplierDiversity defined."); 
+                     }
                  }
-             }
+            }
+        } else if (paymentWorksFormModeService.shouldUseForeignFormProcessingMode()) {
+            LOG.info("buildVendorDiversitiesFromPmwFormCheckboxes, " + PaymentWorksConstants.FOREIGN_FORM_PROCESSING_NOT_IMPLEMENTED_LOG_MESSAGE);
```

Comment:
Would it make sense to have an empty stub method here for processing in case of foreign form mode?